client/chain/chain.go

```diff
@@ -917,9 +979,19 @@ func (c *chainClient) SpotOrder(defaultSubaccountID eth.Hash, network common.Net
 	}
 }
 
+// Deprecated: Use CreateDerivativeOrder instead
 func (c *chainClient) DerivativeOrder(defaultSubaccountID eth.Hash, network common.Network, d *DerivativeOrderData) *exchangetypes.DerivativeOrder {
 
-	market, isPresent := c.marketsAssistant.AllDerivativeMarkets()[d.MarketId]
+	assistant, err := NewMarketsAssistant(network.Name)
```

Comment:
Same concern as spot

Reply:
Added logic in the markets_assistant.go module to make the legacy markets assistant (initialized using the .ini files) a singleton with lazy initialization (to not create it if the user is not using it)